crates/core/src/node/inner/node_executor.rs

```diff
@@ -413,28 +416,49 @@ impl NodeExecutorHandle {
     ///
     /// It is sender's responsibility to make sure [`TxBatch`] is constructed correctly (see its
     /// docs).
-    pub async fn seal_block(&self, tx_batch: TxBatch) -> anyhow::Result<()> {
-        Ok(self
-            .command_sender
+    pub async fn seal_block(&self, tx_batch: TxBatch) -> Result<(), AnvilNodeError> {
+        let block_transactions_hashes = format!("{:?}", tx_batch.txs);
+        self.command_sender
             .send(Command::SealBlock(tx_batch, None))
-            .await?)
+            .await
+            .map_err(|inner| anvil_zksync::node::SealingBlockFailed {
+                block_transactions_hashes,
+                details: inner.to_string(),
+            })
     }
 
     /// Request [`NodeExecutor`] to seal a new block from the provided transaction batch. Waits for
     /// the block to be produced and returns its number.
     ///
     /// It is sender's responsibility to make sure [`TxBatch`] is constructed correctly (see its
     /// docs).
-    pub async fn seal_block_sync(&self, tx_batch: TxBatch) -> anyhow::Result<L2BlockNumber> {
+    pub async fn seal_block_sync(
+        &self,
+        tx_batch: TxBatch,
+    ) -> Result<L2BlockNumber, AnvilNodeError> {
         let (response_sender, response_receiver) = oneshot::channel();
+
+        let debug_transactions_repr = format!("{:?}", tx_batch.txs);
+
         self.command_sender
             .send(Command::SealBlock(tx_batch, Some(response_sender)))
             .await
-            .map_err(|_| anyhow::anyhow!("failed to seal a block as node executor is dropped"))?;
+            .map_err(|_| zksync_error::anvil_zksync::node::SealingBlockFailed {
+                block_transactions_hashes: debug_transactions_repr.clone(),
+                details: "Inner error while sending the block seal request.".to_string(),
+            })?;
```

Comment:
This an other similar places are off. The original message was correct in that there are only one scenario where this can happen - channel is closed meaning node executor has already finished running (due to another error that got propagated from the main execution loop). So *this* particular error has nothing to do with block sealing per se.

Reply:
Fixed in https://github.com/matter-labs/anvil-zksync/commit/d6c874a8036dfe6a32d3d2c1305938ae7fc2e3ed